lib/using_restricted_array.rb

```diff
@@ -9,53 +11,109 @@
 # Time complexity: ?
 # Space complexity: ?
 def length(array)
-  raise NotImplementedError
+  # array[index]
+  i = 0
+  i += 1 until array[i].nil?
+
+  i
```

Comment:
Just for readability I suggest using `return`

```suggestion
  return i
```